plugins/ltac2/tac2intern.ml

```diff
@@ -2059,78 +2056,43 @@ let rec subst_rawexpr subst ({loc;v=tr} as t) = match tr with
 
 (** Registering *)
 
-let () =
+let genintern_core ist locals expected v =
   let open Genintern in
-  let intern ist (ids, tac) =
-    let ids = List.map (fun { CAst.v = id } -> id) ids in
-    let env = match Genintern.Store.get ist.extra ltac2_env with
+  let env = match Genintern.Store.get ist.extra ltac2_env with
     | None ->
       (* Only happens when Ltac2 is called from a toplevel ltac1 quotation *)
       empty_env ~strict:ist.strict_check ()
     | Some env -> env
-    in
-    let fold env id =
-      push_name (Name id) (monomorphic (GTypRef (Other t_ltac1, []))) env
-    in
-    let env = List.fold_left fold env ids in
-    let loc = tac.loc in
-    let (tac, t) = intern_rec env None tac in
-    let () = check_unused_variables ?loc env (List.map (fun x -> Name x) ids) in
-    let () = check_elt_unit loc env t in
-    (ist, (ids, tac))
   in
-  Genintern.register_intern0 wit_ltac2in1 intern
+  let env = List.fold_left (fun env (na,t) -> push_name na t env) env locals in
+  let loc = v.CAst.loc in
+  let v, t = intern_rec env expected v in
+  let () = check_unused_variables ?loc env (List.map fst locals) in
```

Comment:
the test suite is failing because we don't want to always check_unused_variables
(because `Notation foo x := ltac2:(exact 0) + x` would think `x` is unused)